src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -316,12 +316,15 @@ private void VerbatimBlock()
                 Span.ChunkGenerator = SpanChunkGenerator.Null;
             }
 
-            if (!At(CSharpSymbolType.WhiteSpace) && !At(CSharpSymbolType.NewLine))
+            if (IsNested)
             {
-                PutCurrentBack();
+                CompleteBlock(insertMarkerIfNecessary: false);
```

Comment:
In the nested situation we're not `null`ing out the `SpanKind.Code`'s `ChunkGenerator`. I think this should be fine but we should validate with debugging (make sure not excess whitespace is highlighted) once a WTE is produced.